File: pinyougou_seckill_web/src/main/webapp/js/service/payService.js
//服务层
app.service('payService',function($http){

	this.createNative=function(){
		return $http.get("./pay/creatNative.do");
	}

	//查询支付状态:
	this.queryPayStatus=function(out_trade_no){
		return $http.get("./pay/queryPayStatus.do?out_trade_no="+out_trade_no);
	}

});

File: pinyougou_pay_interface/src/main/java/com/pinyougou/pay/service/impl/PayService.java
package com.pinyougou.pay.service.impl;

import com.pinyougou.pojo.TbPayLog;

import java.util.Map;

public interface PayService {

    public abstract Map<String,Object> createNative(String out_trade_no,String total_fee) throws Exception;

    Map<String, String> queryPayStatus(String out_trade_no) throws Exception;

    public TbPayLog getPayLogInRedis(String userId);

    public void updatePayStatusInLog(String out_trade_no, String transaction_id);
}
